model/job-management/src/main/proto/beam_artifact_api.proto

```diff
@@ -102,13 +99,19 @@ message ArtifactChunk {
   bytes data = 1;
 }
 
+message PutArtifactMetadata {
+  // (Required) An identifier for artifact staging session.
+  string artifact_staging_id = 1;
```

Comment:
As far as I can tell, `artifact_staging_id` seems functionally identical to `staging_token` in  `CommitManifestResponse`, and the two names have the same semantic meaning.  I think we should standardize on either either `artifact_staging_id` or `staging_token` as our name for artifact addressing, and stick with it through the entire RPC call chain.

Reply:
I agree the name seems to be very similar but artifact_staging_id and staging_token are fundamentally different. 
artifact_staging_id defines and id to stage token with. This id is generated by JobService. This id can define some meta data to about the artifact like jobId, staging dir etc.
staging_token is a token used to retrieve staged token by ArtifactRetrivalService. This token is generated by ArtifactStagingService.

Reply:
Okay, so then `artifact_staging_id` is in fact more than an identifier.  At the very least we should reflect that in the comments. I guess I can understand that staging and retrieval will require different metadata, but in that case I think we need better, more clear names.  

I think that `artifact_staging_id -> artifact_staging_token` (because it's for staging) and `staging_token -> artifact_retrieval_token` (because it's for retrieval) correctly communicates how they are intended to be used, but it will be confusing for current developers who will hear `artifact_staging_token` and think of the thing that is now called `artifact_retrieval_token`.  An alternative would be `artifact_staging_id -> staging_session_token`, making the distinction that one token is for the staging session, while `staging_token` is the output (which could still have a better name, but maybe we don't fight that battle here).

What do you think?

Reply:
It makes sense. And thanks was giving it a better name.
I will go with `artifact_staging_id -> staging_session_token`

Reply:
+1 for cleaning this up now. Since this is all new and unreleased, there should be very limited impact/confusion.

Reply:
The discussion document alludes to the following items to be specified by the job service and passed to the staging service:

- Base directory to put artifacts in.
- TTL for the artifacts.
- Authentication to submit artifacts.
- Credentials to store artifacts in distributed file system.

Of these, how much is going to fit into the `artifact_staging_id`?  Since this is already a metadata proto, why are we packing data into a string to be parsed later?  Is there a particular parser for it that we already have an implementation for?  If so, we need to document it as such.  Otherwise I'd recommend that any metadata contained within the `artifact_staging_id` should be made explicit as fields in `PutArtifactMetadata`.

Reply:
Given the diversity of ArtifactStagingService, its really hard to capture all the required metadata. 
One thing which we are trying to do in the proto is not to enforce any implementation details. 
From proto perspective, artifact_staging_id/token is just a text string. It can very well be just an id and the implementation of the service can look up that id in another system or it can just be json so that implementation can extract all the relevant information from the token without referring to another system.
This gives the implementation flexibility to implement the service in the way they want.

Reply:
Okay.  Understandable.  Have we considered using an anonymous Struct for implementatation-specific metadata, so that we can decouple it from the identifier?

Reply:
We have a bunch of places where there is an token/id and a runner may choose to put something  there. Migrating to struct for all the tokens is a valid discussion to have but I feel should be separate from this PR as this PR already copies existing behavior when it comes to tokens.

Reply:
I see.  I guess I didn't realize it was that widespread.  In that case I retract my objections, since you're right that we should follow the prevailing style.

Reply:
Just to reiterate on naming. 
I don't have strong preference for naming between `artifact_staging_id and staging_session_token`
But I will go ahead with `staging_session_token`.

Reply:
LGTM